lua/advanced_git_search/utils/table.lua

```diff
@@ -12,4 +12,14 @@ M.index_of = function(array, value)
     return -1
 end
 
+M.merge_table = function(orig, new)
```

Comment:
You might take a look at builtin neovim lua utils https://neovim.io/doc/user/lua.html#vim.tbl_extend() 
Maybe it already does what you need.

Reply:
Good to know, I'm not really familiar with these built-in utils functions.
But apparently, I don't use this `merge_table` function, so I removed it. I'll use the `vim.tbl_extend()` if I ever need it again.

Thanks for the review!